### PR TITLE
Validate loan fillability before filling the loan

### PR DESCRIPTION
**Pull request by jmul-dev**
https://www.pivotaltracker.com/story/show/156739194

Check for token allowance and sufficient balance before filling the loan. If not allowed or insufficient balance, display error toast. Otherwise, display confirmation modal and fill the loan.

### CODE REVIEW

File: src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx [hunk 1]
```diff
@@ -211,6 +212,33 @@ class FillLoanEntered extends React.Component<Props, States> {
         browserHistory.push("/dashboard");
     }
 
+    validateLoanFillability() {
+        try {
+            this.props.handleSetError("");
```
Comment (chrismin) [inline src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx]: Why are we presenting an empty error here?

Comment (chrismin) [reply src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx]: As mentioned offline, let's use `clearToast()` instead!

**Comment by jmul-dev** [comment]
@chrismin This is fixed. I also go through our codebase and replace `setError('')` with `clearToast()` for better code readability

File: src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx [hunk 2]
```diff
@@ -211,6 +213,33 @@ class FillLoanEntered extends React.Component<Props, States> {
...
+            });
+            let error: string = "";
+            if (!tokenToFill) {
+                error = `${principalTokenSymbol} is currently not supported`;
+            } else if (
+                !tokenToFill.tradingPermitted ||
+                (debtOrder.principalAmount && tokenToFill.balance.lt(debtOrder.principalAmount))
+            ) {
+                error = "Creditor allowance is insufficient";
+            }
```
Comment (nadavhollander) [inline src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx]: This seems like an okay stop-gap solution, but the more comprehensive solution would be to add an endpoint to `dharma.js` for validating loan fillability.  We can merge this for now, though, and I'll create another ticket under `must-have` for a more comprehensive solution in `dharma.js`

Comment (nadavhollander) [reply src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx]: @chrismin FYI

Comment (nadavhollander) [reply src/modules/FillLoan/FillLoanEntered/FillLoanEntered.tsx]: Actually, I take that back -- I think this PR is a misinterpretation of the Pivotal Task.  These checks already occur in `fillAsync` in dharma.js.  There are, however, other validation checks that aren't being done -- namely, things specific to collateralization (*i.e. is the debtor have sufficient collateral token balance / allowance?).  I still think this is best done in `dharma.js`.  Closing and rejecting PT task.

**Pull request closed without merging**